src/org/labkey/test/components/core/ApiKeyDialog.java

```diff
@@ -27,7 +28,10 @@ public ApiKeyDialog(WebDriver driver, String title)
     public ApiKeyDialog generateApiKey()
     {
         elementCache().generateApiKeyButton.click();
-        return new ApiKeyDialog(getDriver(), _title);
+        getWrapper().shortWait().until(ExpectedConditions.invisibilityOf(elementCache().descriptionInput.getComponentElement()));
+        clearElementCache();
+        getWrapper().shortWait().until(ExpectedConditions.visibilityOf(elementCache().inputField.getComponentElement()));
+        return this;
```

Comment:
I would have thought using `this` might produce a stale element exception. If not, then so much the better.

Reply:
The body of the dialog does go stale (hence the `clearElementCache`) but the outer div of the dialog is stable.